### PR TITLE
Support ISO 8601 Format in Date Format.

### PR DESCRIPTION
**Pull request by vamsi-amazon**
Signed-off-by: vamsi-amazon <reddyvam@amazon.com>

### Description
Supported ISO 8601 Format in Date Format. IN order to solve this we are allowing all the alphabet characters with a '%' character behind to be replicated into the output format as is.
Eg: 
%Y-%m-%dT%TZ   --- 2022-01-15T11:12:29Z

 
### Issues Resolved
https://github.com/opensearch-project/sql/pull/434

 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java [hunk 1]
```diff
@@ -123,7 +123,10 @@ public void setup() {
       new DateFormatTester("2008-12-31",
           ImmutableList.of("%v","%V","%u","%U"),
           ImmutableList.of("53","52","53","52")
-      )
+      ),
+      new DateFormatTester("1998-01-31 13:14:15.012345",
```
Comment (penghuo) [inline core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java:127]: add not-match cases? e.g. "%Y-%m-%da%Ta"

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFormatterUtil.java [hunk 2]
```diff
@@ -79,6 +79,8 @@
       .build();
 
   private static final Pattern pattern = Pattern.compile("%.");
+  private static final Pattern CHARACTERS_WITH_NO_MOD_LITERAL_BEHIND_PATTERN
+          = Pattern.compile("(?<!%)[a-zA-Z&&[^aydmshiHIMYDSEL]]+");
```
Comment (penghuo) [inline core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFormatterUtil.java:83]: Update docs,
1. SQL, https://github.com/opensearch-project/sql/blob/main/docs/user/dql/functions.rst#date-format
2. PPL, https://github.com/opensearch-project/sql/blob/main/docs/user/ppl/functions/datetime.rst

File: core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java (hunk 1 above)
Comment (vamsi-amazon) [reply core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java:127]: Added few more testcases.

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFormatterUtil.java (hunk 2 above)
Comment (vamsi-amazon) [reply core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFormatterUtil.java:83]: Done.

**Review by penghuo — APPROVED** [review]
LGTM

**Pull request merged**